lib/secret_keys/cli.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
+
+require "optparse"
+require "io/console"
+
+require_relative "../secret_keys.rb"
+
+class SecretKeys
+  class CLI
+    class Encrypt < CLI
+      def parse_additional_options(opts)
+        @new_secret_key = nil
+        opts.on("--new-secret-key ENCRYPTION_KEY", String, "Encryption key used to encrypt strings in the file on output. This option can be used to change the encryption key.") do |value|
+          @new_secret_key = value
+        end
+      end
+
+      def run
+      end
+    end
+
+    class Decrypt < CLI
+      def run
+      end
+    end
+
+    class Read < CLI
+      def parse_additional_options(opts)
+        @key = nil
+        opts.on("-k", "--key KEY", String, "Key from the file to output. You can use dot notation to read a nested key.") do |value|
+          @key = value
+        end
+      end
+
+      def run
+        raise ArgumentError.new("key is required") if @key.nil? || @key.empty?
+        val = secrets.to_h
+        @key.split(".").each do |key|
+          val = secrets[key] if val.is_a?(Hash)
+        end
+        val
+      end
+    end
+
+    class Edit < CLI
+      def parse_additional_options(opts)
+        @actions = []
+        opts.on("-e", "--set-encrypted KEY=VALUE", String, "Set an encrypted value in the file. You can use dot notation to set a nested value.") do |value|
+          key, val = value.split("=", 2)
+          @actions << [:encrypt, key, val]
+        end
+        opts.on("-d", "--set-decrypted KEY=VALUE", String, "Set a plain text value in the file. You can use dot notation to set a nested value.") do |value|
+          key, val = value.split("=", 2)
+          @actions << [:decrypt, key, val]
+        end
+        opts.on("-r", "--set-encrypted KEY=VALUE", String, "Remove a key from the file. You can use dot notation to remove a nested value.") do |value|
+          @actions << [:remove, value]
+        end
+      end
+
+      def run
+      end
+    end
+
+    attr_reader :secrets
+
+    def initialize(argv)
+      parse_options(argv)
+      @secrets = SecretKeys.new(@input, @secret_key)
+    end
+
+    # Subclasses can override this method to parse additional options beyond the standard set.
+    def parse_additional_options(opts)
+    end
+
+    # Subclasses must implement this method to execute the logic.
+    def run
+      raise NotImplementedError
+    end
+
+    private
+
+    def parse_options(argv)
+      @secret_key = nil
+      @format = nil
+
+      OptionParser.new do |opts|
+        opts.banner = "Usage: secret_keys encrypt|decrypt|read|edit [options] input_file|- output_file|-"
```

Comment:
```suggestion
        opts.banner = "Usage: secret_keys <encrypt|decrypt|read|edit> [options] [--] INFILE [OUTFILE|-]"
```
more clearly indicate required/optional params. Specifying stuff like the default for infile should be done under a separate annotation. Not sure if you also want to have printing a global help be this, can then subcommands will only print the help for that command. For example:

```
$ gem --help
RubyGems is a sophisticated package manager for Ruby.  This is a
basic help message containing pointers to more information.

  Usage:
    gem -h/--help
    gem -v/--version
    gem command [arguments...] [options...]

  Examples:
...

$ gem install --help
Usage: gem install GEMNAME [GEMNAME ...] [options] -- --build-flags [options]

  Options:
        --platform PLATFORM          Specify the platform of gem to install
    -v, --version VERSION            Specify version of gem to install
...
```